File: RusticShop/src/app/shared/models/VProductList.ts
export default interface VProductList {
  id: number;
  name: string;
  shortDescription: string;
  description?: string;
  unitPrice: number;
  categories?: string;
  features?: string;
  imagesCount: string;
  discountsCount: string;
  isPublished: boolean;
}

export interface VProductListFeature {
  name: string;
  content: string;
}

export function getVProductListCategories(item: VProductList): string[] {
  return item.categories?.split(',') ?? [];
}

export function getFeatures(item: VProductList): VProductListFeature[] {
  return item.features?.split(';').map( feature => {
    const [name, content] = feature.split(':');
    return { name, content };
  }) ?? [];
}

File: RusticShopAPI/Data/Models/Views/VProductList.cs
﻿using System.ComponentModel.DataAnnotations.Schema;

namespace RusticShopAPI.Data.Models.Views
{
    public class VProductList
    {
        public long Id { get; set; }
        public string Name { get; set; } = null!;
        public string ShortDescription { get; set; } = null!;
        public string? Description { get; set; } = null!;

        [Column(TypeName = "MONEY")]
        public decimal UnitPrice { get; set; }
        public string? Categories { get; set; } = null!;
        public string? Features { get; set; } = null!;
        public int ImagesCount { get; set; }
        public int DiscountsCount { get; set; }
        public bool IsPublished { get; set; }
    }
}
